src/5.Functions/functions.php

```diff
@@ -33,6 +33,8 @@ function object_generate()
     }
     $anObject->arr1 = $anArray;
     $anObject->arr2 = $anAssocArray;
+    $anObject->arr1P1 = array_slice($anArray, 0, floor(count($anArray) / 2), true);
```

Comment:
Having re-read instructions an n-th time, decided that $anArray is "the original array". Whatever!